ledger/src/blocktree.rs

```diff
@@ -252,119 +252,132 @@ impl Blocktree {
     /// Does not check for integrity and does not update slot metas that refer to deleted slots
     /// Modifies multiple column families simultaneously
     pub fn purge_slots(&self, mut from_slot: Slot, to_slot: Option<Slot>) {
-        // split the purge request into batches of 1000 slots
+        // if there's no upper bound, split the purge request into batches of 1000 slots
         const PURGE_BATCH_SIZE: u64 = 1000;
-        let mut batch_end = to_slot
-            .unwrap_or(from_slot + PURGE_BATCH_SIZE)
-            .min(from_slot + PURGE_BATCH_SIZE);
+        let mut batch_end = to_slot.unwrap_or(from_slot + PURGE_BATCH_SIZE);
         while from_slot < batch_end {
-            if let Ok(end) = self.run_purge_batch(from_slot, batch_end) {
-                // no more slots to iter or reached the upper bound
-                if end {
+            match self.run_purge(from_slot, batch_end) {
+                Ok(end) => {
+                    if let Err(e) = self.compact_storage(from_slot, batch_end) {
```

Comment:
@sunnygleason I had something more like this in mind. That combined function got way too large.

Reply:
@sagar-solana I think I understand, but it's not totally clear because this PR does a lot. Would you like me to take a shot at applying the minimal change in a smaller PR?

Reply:
Maybe I can explain. What is not clear?
I basically reintroduced your older function and just called it here. I prefer it to the merged function you introduced.
It's probably not a bad idea for you to review the rest of this pr as well.

Reply:
@sagar-solana sure thing! the main issue is that I can't benchmark this as cleanly because it's not the small/targeted change of `delete` vs `delete_range`

Reply:
I could give you a patch that does it with delete_slot if you want to compare but based on the docs, we shouldn't be doing a range scan and delete. Does that sound fair?

Reply:
sounds great!

Reply:
@sunnygleason https://github.com/solana-labs/solana/pull/7549
Lets see how the perf changes :D

Reply:
I've merged this one for now, if we see that `delete_slot` is faster for some reason, we can easily switch over to it via #7549